test/blockchain.js

```diff
@@ -95,7 +95,15 @@ describe('embark.Blockchain', function () {
           config.rpcPort += constants.blockchain.servicePortOnProxy;
         }
 
-        assert.deepEqual(blockchain.config, config);
+        try {
+            assert.deepEqual(blockchain.config, config);
+        } catch (e) {
+            // Verify Ether unit conversion to wei
+            assert(e instanceof assert.AssertionError);
+            assert.equal(e.actual['targetGasLimit'], '8000000');
+            assert.deepEqual(Object.assign({}, blockchain.config, {targetGasLimit: undefined}),
+                             Object.assign({}, config, {targetGasLimit: undefined}));
+        }
```

Comment:
So the way I see it is that, just to ensure we can still `deepEqual` whatever is being created by `new Blockchain()` and the test `config`, we're introducing a try/catch here, which will always fail (because `8 Mwei` is never `=== 8000000`), making it kind of unecessary.

If we still want to have a test that checks whether the configs are indeed the same I'd propose we remove the try/catch and always unset `targetGasLimit` before we make an assertion. Another option would be to only assert `equal` **some** of the config properties and assume that it probably worked out.

I think it much more important test here would be to ensure that `8 Mwei` indeed ends up as `8000000`, which is what you're doing in line `103`. Since this is a new feature though, I'd personally prefer seeing this as a separate spec.

Reply:
@PascalPrecht sounds good! I'll add the tests as separate specs and do the `deepEqual` minus the Ether values and assert those directly. Thanks for the feedback!